File: backend/src/main/java/org/kmurygin/healthycarbs/mealplan/dto/RecipeDTO.java
package org.kmurygin.healthycarbs.mealplan.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kmurygin.healthycarbs.mealplan.DietType;
import org.kmurygin.healthycarbs.mealplan.MealType;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RecipeDTO {
    private Long id;
    private String name;
    private String description;
    private String instructions;
    private List<RecipeIngredientDTO> ingredients;
    private Double calories;
    private Double carbs;
    private Double protein;
    private Double fat;
    private MealType mealType;
    private DietType dietType;
    private Boolean isFavourite;
    private Long favouritesCount;
}

File: frontend/src/app/core/models/dto/recipe.dto.ts
import type {RecipeIngredientDto} from './recipe-ingredient.dto';
import type {MealType} from "../enum/meal-type.enum";
import type {DietType} from "../enum/diet-type.enum";
import type {UserDto} from "@core/models/dto/user.dto";

export interface RecipeDto {
  readonly id: number;
  readonly name: string;
  readonly description: string;
  readonly instructions: string;
  readonly ingredients: readonly RecipeIngredientDto[];
  readonly calories: number;
  readonly carbs: number;
  readonly protein: number;
  readonly fat: number;
  readonly dietType: DietType;
  readonly mealType: MealType;
  readonly isFavourite: boolean;
  readonly favouritesCount: number;
  readonly author: UserDto | null;
}
